### DISCUSSION TITLE
How to use new dev server with cloudflare pages?

### DISCUSSION THREAD

**Discussion by kishanhitk** [remix-run/remix#5462] [Q&A] [answered]
I want to use the new dev server on the Cloudflare pages template built using Remix 1.13.0
I did the following 
1. Add `future: {
    unstable_dev: true,
  },` to `remix.config.js`
2. Changed `"dev:remix": "remix watch",` to `"dev:remix": "remix dev",` in `package.json` scripts.

Now, whenever I change and save something, I get the message in console
```
💿 File changed: app/routes/index.tsx
💿 Rebuilding...
💿 Rebuilt in 117ms
```

But the browser does not automatically refresh to show the changes. If I refresh the browser manually it now shows the updated changes. Ideally, the browser should refresh and show new changes automatically once `Rebuilt`. Can anyone point out what I am missing in my config?

My remix.config.js
```
module.exports = {
  devServerBroadcastDelay: 1000,
  ignoredRouteFiles: ["**/.*"],
  server: "./server.js",
  serverBuildPath: "functions/[[path]].js",
  serverConditions: ["worker"],
  serverDependenciesToBundle: "all",
  serverMainFields: ["browser", "module", "main"],
  serverMinify: true,
  serverModuleFormat: "esm",
  serverPlatform: "neutral",
  future: {
    unstable_dev: true,
  },
  // appDirectory: "app",
  // assetsBuildDirectory: "public/build",
  // publicPath: "/build/",
};
```

My package.json script
```
  "scripts": {
    "build": "remix build",
    "dev:remix": "remix dev",
    "dev:wrangler": "cross-env NODE_ENV=development npm run wrangler",
    "dev": "npm-run-all build --parallel \"dev:*\"",
    "start": "cross-env NODE_ENV=production npm run wrangler",
    "typecheck": "tsc",
    "wrangler": "wrangler pages dev ./public"
  },
```

**Comment by pcattori** [comment]
With the new dev server flow, its the responsibility of the script running your _app server_ (i.e. `wrangler`) to pick up changes to `build/index.js`. With `node` 18, you could use the `--watch` flag. Prior versions of `node` can use `nodemon`.

**This will change and be simplified in v1.16**

I'm not well-versed with `wrangler`, but you'll need to figure out how to do the equivalent thing. Does `wrangler` have a way to watch for changes?

**Comment by kishanhitk** [comment, marked as answer]
Thanks, @pcattori, I found the solution.
We have to make changes at two places:
1. Add `--live-reload` in wrangler command. So it now becomes `"wrangler": "wrangler pages dev --live-reload ./public"`.
2. Pass `appServerPort` in `unstable_dev` option, which is `8788` by default for wrangler. So, it now becomes 
              ``` 
                              unstable_dev: {
                                          appServerPort: 8788,
                                      },
              ```

**Comment by ChristianJacobsen** [comment]
I had to omit `--live-reload` for HMR to work. With `--live-reload` I kept getting full HTTP GET requests for the route, but without it, I could see the expected streamed updates of JS.

**Reply by philippeaka** [reply]
I had the same experience, thanks
